File: src/com/jda/functionalprograms/Distance.java
package com.jda.functionalprograms;

import com.jda.utility.Maths;
import com.jda.utility.Printer;
import com.jda.utility.Validation;

public
class Distance {

  public static
  void main(String[] args) {
    if (!Validation.checkCmdArgs(args, 2)) { return; }
    double x = Double.parseDouble(args[0]);
    double y = Double.parseDouble(args[1]);
    Printer.printLine(
        "The distance of (" + x + "," + y + ") from origin is " + distanceFromOrigin(x, y));
  }

  private static
  double distanceFromOrigin(double x, double y) {
    return Maths.euclideanDistance(x, y, 0.0, 0.0);
  }
}

File: src/com/jda/functionalprograms/Distance.kt
package com.jda.functionalprograms

import com.jda.utility.Utility

object Distance {

    @JvmStatic
    fun main(args: Array<String>) {
        if (!Utility.checkCmdArgs(args, 2)) {
            return
        }
        val x = java.lang.Double.parseDouble(args[0])
        val y = java.lang.Double.parseDouble(args[1])
        Utility.printLine("The distance of (" + x + "," + y + ") from origin is " + distanceFromOrigin(x, y))
    }

    private fun distanceFromOrigin(x: Double, y: Double): Double {
        return Utility.euclideanDistance(x, y, 0.0, 0.0)
    }
}
